evalml/automl/automl_algorithm/evalml_algorithm.py

```diff
@@ -0,0 +1,365 @@
+            next_batch = self._create_ensemble()
+        elif self.batch_number == 4:
+            next_batch = self._create_long_top_n(n=3)
+        elif self.batch_number % 2 != 0:
+            next_batch = self._create_ensemble()
+        else:
+            next_batch = []
+            for _ in range(10):
+                for pipeline in self._top_n_pipelines:
+                    proposed_parameters = self._tuners[pipeline.name].propose()
+                    parameters = self._transform_parameters(
+                        pipeline, proposed_parameters
+                    )
+                    parameters.update(
+                        {"Select Columns Transformer": {"columns": self._selected_cols}}
+                    )
+                    next_batch.append(
+                        pipeline.new(
+                            parameters=parameters, random_seed=self.random_seed
+                        )
+                    )
+
+        self._pipeline_number += len(next_batch)
+        self._batch_number += 1
+        return next_batch
+
+    def add_result(self, score_to_minimize, pipeline, trained_pipeline_results):
+        """Register results from evaluating a pipeline
+
+        Arguments:
+            score_to_minimize (float): The score obtained by this pipeline on the primary objective, converted so that lower values indicate better pipelines.
+            pipeline (PipelineBase): The trained pipeline object which was used to compute the score.
+            trained_pipeline_results (dict): Results from training a pipeline.
+        """
+        if pipeline.model_family != ModelFamily.ENSEMBLE:
+            if self.batch_number >= 3:
+                super().add_result(
+                    score_to_minimize, pipeline, trained_pipeline_results
+                )
+
+        if self.batch_number == 2 and self._selected_cols is None:
+            if is_regression(self.problem_type):
+                self._selected_cols = pipeline.get_component(
+                    "RF Regressor Select From Model"
+                ).get_names()
+            else:
+                self._selected_cols = pipeline.get_component(
+                    "RF Classifier Select From Model"
+                ).get_names()
+
+        current_best_score = self._best_pipeline_info.get(
+            pipeline.model_family, {}
+        ).get("mean_cv_score", np.inf)
+        if (
+            score_to_minimize is not None
+            and score_to_minimize < current_best_score
+            and pipeline.model_family != ModelFamily.ENSEMBLE
+        ):
+            self._best_pipeline_info.update(
+                {
+                    pipeline.model_family: {
+                        "mean_cv_score": score_to_minimize,
+                        "pipeline": pipeline,
+                        "parameters": pipeline.parameters,
+                        "id": trained_pipeline_results["id"],
+                    }
+                }
+            )
```

Comment:
I'm not sure how I feel about all this being in `add_result()`, not like you really have much other choice, but I think maybe some of what's happening might want to make its way into the docstring.  I'm not 100% sure what _selected_cols is, or why it's being updated and it's also good to know that the _best_pipeline_info is being updated here.